### PR TITLE
[WIP] Refactor/develop

### PR DESCRIPTION
**Pull request by satellitex**
# Purpose
- `PeerServiceConfig` to migrate `Hijiri`. ( [WIP] Plase read hijiri.hpp )
- Review directory structure.
- Clean the contents of `util`. and `util` rename `utils`.

### CODE REVIEW

**Comment by satellitex** [comment]
Plesase check `core/service/peer_service.hpp` and `core/service/hijiri.hpp`
Is `peer_service.hpp` too fat?

I think
- Peer :: Node should be written in a separate file.
- directory structure have to  change to that
```
core - peer - hijiri
                    L izanami
                    L service
                    L node
                    L etc..
```

**Comment by takemiyamakoto** [comment]
circle-ciを通れるようにしたらいい

**Comment by l4l** [comment]
that pretty hard to review, lets merge #209 before
then apply your fixes to the top branch commit

**Comment by l4l** [comment]
Could you please save your fixes and recreate this branch from current develop branch?
There's lots of conflicts

File: core/service/izanami.cpp [hunk 1]
```diff
@@ -22,160 +22,216 @@ limitations under the License.
...
-    using Api::TransactionResponse;
 
-    InitializeEvent::InitializeEvent() {
-        now_progress = 0;
-    }
+        void setAwkTimer(int const sleepMillisecs, const std::function<void(void)> &action) {
+            std::thread([action, sleepMillisecs]() {
+                std::this_thread::sleep_for(std::chrono::milliseconds(sleepMillisecs));
+                action();
+            }).join();
```
Comment (satellitex) [inline core/service/izanami.cpp]: We refactor `utils/`, genreral function will migrate to there.
Would you wait ?

**Comment by motxx** [comment]
I've got some trouble with new circle-ci system when merging develop.
https://circleci.com/gh/hyperledger/iroha/1860?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

@Warchant 
Could you look through circle-ci-link?

**Comment by Warchant** [comment]
Ah, I see. WIP.

**Comment by motxx** [comment]
Thanks for your fixing!

**Comment by Warchant** [comment]
*UPD*, please pull again, there is one more fix.

**Pull request merged**